starport/services/common.go

```diff
@@ -0,0 +1,64 @@
+package services
+
+import (
+	"path/filepath"
+
+	starportconf "github.com/tendermint/starport/starport/chainconf"
+	"github.com/tendermint/starport/starport/pkg/cosmosanalysis/module"
+	"github.com/tendermint/starport/starport/pkg/cosmosgen"
+	"github.com/tendermint/starport/starport/pkg/giturl"
+)
+
+// GetClientGenerationOptions gets client generation options.
+func GetClientGenerationOptions(
+	projecPath string,
+	goModPath string,
+	enableThirdPartyModuleCodegen bool,
+	conf starportconf.Config,
+) []cosmosgen.Option {
+	options := []cosmosgen.Option{
+		cosmosgen.WithGoGeneration(goModPath),
+		cosmosgen.IncludeDirs(conf.Build.Proto.ThirdPartyPaths),
+	}
+
+	// generate Vuex code as well if it is enabled.
+	if conf.Client.Vuex.Path != "" {
+		storeRootPath := filepath.Join(projecPath, conf.Client.Vuex.Path, "generated")
+		options = append(options,
+			cosmosgen.WithVuexGeneration(
+				enableThirdPartyModuleCodegen,
+				getModulePathFn(storeRootPath),
+				storeRootPath,
+			),
+		)
+	}
+
+	// generate ts client code as well if it is enabled.
+	// NB: Vuex generates JS/TS client code as well but in addtion to a vuex store.
+	// Path options will conflict with each other.
+	if conf.Client.Typescript.Path != "" {
+		tsRootPath := filepath.Join(projecPath, conf.Client.Typescript.Path)
+		options = append(options,
+			cosmosgen.WithJSGeneration(
+				enableThirdPartyModuleCodegen,
+				getModulePathFn(tsRootPath),
+			),
+			// Only generate typescript client files.
+			cosmosgen.WithTSGeneration(),
```

Comment:
```suggestion
```

I think we should drop this and replace WithJSGeneration to WithTSGeneration. Otherwise, it is a bit less abovious that it'll only generate TS files.

Reply:
Sounds good, doesn't look like `WithJSGeneration` is used anywhere else so should be a safe change.